datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -0,0 +1,506 @@
+            right,
+        } if is_false(right) => simplify(left),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::Or,
+            right,
+        } if left == right => simplify(left),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::And,
+            right,
+        } if is_false(left) || is_false(right) => lit(false),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::And,
+            right,
+        } if is_true(right) => simplify(left),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::And,
+            right,
+        } if is_true(left) => simplify(right),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::And,
+            right,
+        } if left == right => simplify(right),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::Multiply,
+            right,
+        } if is_one(left) => simplify(right),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::Multiply,
+            right,
+        } if is_one(right) => simplify(left),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::Divide,
+            right,
+        } if is_one(right) => simplify(left),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::Divide,
+            right,
+        } if left == right => lit(1),
+        Expr::BinaryExpr { left, op, right }
+            if left == right && operator_is_boolean(*op) =>
+        {
+            simplify(left)
+        }
+        Expr::BinaryExpr {
+            left,
+            op: Operator::Or,
+            right,
+        } if expr_contains(left, right) => as_binary_expr(left)
+            .map(|x| match x {
+                Expr::BinaryExpr {
+                    left: _,
+                    op: Operator::Or,
+                    right: _,
+                } => x.clone(),
+                Expr::BinaryExpr {
+                    left: _,
+                    op: Operator::And,
+                    right: _,
+                } => *right.clone(),
+                _ => expr.clone(),
+            })
+            .unwrap_or(expr.clone()),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::Or,
+            right,
+        } if expr_contains(right, left) => as_binary_expr(right)
+            .map(|x| match x {
+                Expr::BinaryExpr {
+                    left: _,
+                    op: Operator::Or,
+                    right: _,
+                } => *right.clone(),
+                Expr::BinaryExpr {
+                    left: _,
+                    op: Operator::And,
+                    right: _,
+                } => *left.clone(),
+                _ => expr.clone(),
+            })
+            .unwrap_or(expr.clone()),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::And,
+            right,
+        } if expr_contains(left, right) => as_binary_expr(left)
```

Comment:
Also here - is this still needed and doesn't this remove certain oppurtunities as the recursive call isn't done?